test/test_transforms_v2_refactored.py

```diff
@@ -0,0 +1,146 @@
+    input_cuda = input_cuda.as_subclass(torch.Tensor)
+    input_cpu = input_cuda.to("cpu")
+
+    actual = kernel(input_cuda, *other_args, **kwargs)
+    expected = kernel(input_cpu, *other_args, **kwargs)
+
+    assert_close(actual, expected)
+
+
+@cache
+def _script(fn):
+    try:
+        return torch.jit.script(fn)
+    except Exception as error:
+        raise AssertionError(f"Trying to `torch.jit.script` '{fn.__name__}' raised the error above.") from error
+
+
+def _check_scripted_vs_eager(kernel_eager, input, *other_args, **kwargs):
+    kernel_scripted = _script(kernel_eager)
+
+    input = input.as_subclass(torch.Tensor)
+    actual = kernel_scripted(input, *other_args, **kwargs)
+    expected = kernel_eager(input, *other_args, **kwargs)
+
+    assert_close(actual, expected)
+
+
+def _unbatch(batch, *, data_dims):
+    if isinstance(batch, torch.Tensor):
+        batched_tensor = batch
+        metadata = ()
+    else:
+        batched_tensor, *metadata = batch
+
+    if batched_tensor.ndim == data_dims:
+        return batch
+
+    return [
+        _unbatch(unbatched, data_dims=data_dims)
+        for unbatched in (
+            batched_tensor.unbind(0) if not metadata else [(t, *metadata) for t in batched_tensor.unbind(0)]
+        )
+    ]
+
+
+def _check_batched_vs_single(kernel, batched_input, *other_args, **kwargs):
+    input_type = datapoints.Image if is_simple_tensor(batched_input) else type(batched_input)
+    # This dictionary contains the number of rightmost dimensions that contain the actual data.
+    # Everything to the left is considered a batch dimension.
+    data_dims = {
+        datapoints.Image: 3,
+        datapoints.BoundingBox: 1,
+        # `Mask`'s are special in the sense that the data dimensions depend on the type of mask. For detection masks
+        # it is 3 `(*, N, H, W)`, but for segmentation masks it is 2 `(*, H, W)`. Since both a grouped under one
+        # type all kernels should also work without differentiating between the two. Thus, we go with 2 here as
+        # common ground.
+        datapoints.Mask: 2,
+        datapoints.Video: 4,
+    }.get(input_type)
+    if data_dims is None:
+        raise pytest.UsageError(
+            f"The number of data dimensions cannot be determined for input of type {input_type.__name__}."
+        ) from None
+    elif batched_input.ndim <= data_dims or not all(batched_input.shape[:-data_dims]):
+        # input is not batched or has a degenerate batch shape
+        return
+
+    batched_input = batched_input.as_subclass(torch.Tensor)
+    batched_output = kernel(batched_input, *other_args, **kwargs)
+    actual = _unbatch(batched_output, data_dims=data_dims)
+
+    single_inputs = _unbatch(batched_input, data_dims=data_dims)
+    expected = tree_map(lambda single_input: kernel(single_input, *other_args, **kwargs), single_inputs)
+
+    assert_close(actual, expected)
+
+
+def check_kernel(
+    kernel,
+    input,
+    *other_kernel_args,
```

Comment:
`other_kernel_args` is called `other_args` in the other helpers. Might be worth keeping the same name everywhere. I'd suggest `kernel_args` for all, as the "other" part isn't really necessary (it's more strictly correct because `input` isn't part of it, but does it matter?)

Reply:
I can align everything.